parser/src/parsers/program.rs

```diff
@@ -1,198 +1,136 @@
-use crate::parse_assign;
+use crate::eat_type;
+use crate::parse_expr;
 use crate::parse_ident;
 use crate::parse_ident_import;
 use crate::parse_ident_type;
 use crate::parse_ident_typed;
-use crate::parse_stmt;
+use crate::parse_ident_typed_strict;
+use crate::parse_stmts;
 use crate::parser::Parser;
 use crate::ParserError;
 use crate::ParserErrorKind;
 
+use whistle_ast::IdentType;
 use whistle_ast::IdentTyped;
+use whistle_ast::Literal;
 use whistle_ast::ProgramStmt;
 use whistle_common::Keyword;
+use whistle_common::Operator;
+use whistle_common::Primitive;
 use whistle_common::Punc;
 use whistle_common::Token;
 
 pub fn parse_program(parser: &mut Parser) -> Result<ProgramStmt, ParserError> {
-  match parser.peek() {
-    Some(Token::Keyword(Keyword::Fun)) | Some(Token::Keyword(Keyword::Export)) => {
-      parse_fun_decl(parser)
+  match parser.peek()? {
+    Token::Keyword(Keyword::Fun) | Token::Keyword(Keyword::Export) => parse_fun_decl(parser),
+    Token::Keyword(Keyword::Import) => parse_import(parser),
+    Token::Keyword(Keyword::Val) => parse_val_decl(parser),
+    Token::Keyword(Keyword::Var) => parse_var_decl(parser),
+    Token::Keyword(Keyword::Struct) => parse_struct_decl(parser),
```

Comment:
Note for future: we will want to be able to export global variables, re-export imports and structs